pavelib/quality.py

```diff
@@ -147,10 +138,43 @@ def run_pylint(options):
     with open(Env.METRICS_DIR / "pylint", "w") as f:
         f.write(violations_count_str)
 
-    # Fail number of violations is greater than the limit
-    if num_violations > violations_limit > -1:
-        raise BuildFailure("Failed. Too many pylint violations. "
-                           "The limit is {violations_limit}.".format(violations_limit=violations_limit))
+    # Fail when number of violations is less than the lower limit,
+    # which likely means that pylint did not run successfully.
+    # If pylint *did* run successfully, then great! Modify the lower limit.
+    if num_violations < lower_violations_limit > -1:
+        raise BuildFailure(
+            "Failed. Too few pylint violations. "
+            "Expected to see at least {lower_limit} pylint violations. "
+            "Either pylint is not running correctly -or- "
+            "the lower limit should be lowered or removed.".format(lower_limit=lower_violations_limit)
```

Comment:
Maybe this is a nit, but I don't want someone to blindly follow this without thinking about it just to get the build to pass.  If the lower limit needs to move, the upper limit should be move too. It should read:
```
"the limits should be lowered and/or the lower limit should be removed.".format(lower_limit=lower_violations_limit)
```
Of course, this question around removing the lower limit makes me wonder about adding a fail safe again.  Why not force a lower limit of 1 (if not supplied) when the upper limit is greater than 0 to protect against a bad config allowing this problem to happen all over again?

Reply:
Sorry if this is being overly nit-picky.  Thank you for adding this protection.

Reply:
I changed the text to what you recommended.
This PR implements/supplies a lower limit for the main edx-platform repo and its Jenkins quality testing to protect against broken pylint testing. I'm happy to review a follow-on PR that adds more complexity or fail-safe checking - but I'd like to merge this one to get pylint checking happening again.

Reply:
No worries.  Just throwing out ideas, but I agree, and don't want to be a blocker.  I wasn't doing a full-review, so just merge on @nedbat's approval.